### PR TITLE
Changed endpoint for reading messages

### PR DESCRIPTION
**Pull request by alexlongerbeam**
## Description
Changed endpoint for reading messages to read all messages in a specified thread

## Pls do 
- [x] Tested and working
- [x] Unit tests

## Test plan
Showing thread with multiple unread messages:

screen shot 2018-04-30 at 3 57 16 pm

PATCH to /messaging/read/<THREAD_ID>

screen shot 2018-04-30 at 4 00 34 pm

Showing same thread, with all messages read:

screen shot 2018-04-30 at 4 00 51 pm

### CODE REVIEW

**Comment by alexlongerbeam** [comment]
I'm pretty sure I fixed all the merge conflicts with web sockets because there are some methods used in master that aren't defined in my branch yet so I can't fully test them. But should all work

**Comment by ramsgoli** [comment]
shouldn't the `unread` key now be on the thread object instead of the message object?

**Comment by alexlongerbeam** [comment]
I thought we decided to keep individual messages tracked on read or unread, but just change the endpoint to mark threads read. When you get a thread it includes the latest message which says read or unread

File: src/messaging/views.py [hunk 1]
```diff
@@ -35,25 +35,27 @@ def websockets_notify_user(user):
...
-
-        message.unread = False
+        thread_id = int(self.kwargs['thread_id'])
+        thread = get_object_or_404(Thread, id=thread_id)
+        unread_messages = Message.objects.filter(Q(thread=thread) & Q(unread=True))
 
-        message.save()
+        for message in unread_messages:
+            message.unread = False
+            message.save()
```
Comment (marktai) [inline src/messaging/views.py]: this is a "n+1 query".  you can look up what it is and why it isn't great.

It honestly doesnt matter for our scale, but I think you would learn something by fixing it.  

use the `unread_messages` queryset and update them all at once

**Pull request merged**